### PR TITLE
fix: make nullable as default when alter table

### PR DESCRIPTION
**Pull request by yfractal**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
Fix bug https://github.com/GreptimeTeam/greptimedb/issues/364
Make nullable as default when alter table.

Please explain IN DETAIL what the changes are in this PR and why they are needed:
- Summarize your change (**mandatory**)
   Make nullable as default when alter table
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- []  I have written the necessary rustdoc comments.
- []  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/sql/src/statements.rs [hunk 1]
```diff
@@ -260,10 +260,12 @@ pub fn column_def_to_schema(column_def: &ColumnDef, is_time_index: bool) -> Resu
 pub fn sql_column_def_to_grpc_column_def(col: ColumnDef) -> Result<api::v1::ColumnDef> {
     let name = col.name.value.clone();
     let data_type = sql_data_type_to_concrete_data_type(&col.data_type)?;
-    let nullable = col
-        .options
-        .iter()
-        .any(|o| matches!(o.option, ColumnOption::Null));
+
+    let nullable = col.options.is_empty()
```
Comment (killme2008) [inline src/sql/src/statements.rs:264]: Nullable as the default means we can't find `ColumnOption::NotNull` in `col.options`. This modification is not enough here.

Comment (yfractal) [reply src/sql/src/statements.rs]: it's more clear, updated.

Comment (PTham1993) [reply src/sql/src/statements.rs]: > Nullable as the default means we can't find `ColumnOption::NotNull` in `col.options`. This modification is not enough here.

Cartoons

**Review by waynexia — APPROVED** [review]
Looks good to me. Thanks for your contribution.

Could you please also add some unit tests to cover this scenario? Like asserting when the `ColumnDef` will be nullable and when not.

**Review by killme2008 — APPROVED** [review]
LGTM

**Comment by yfractal** [comment]
Thanks every one.

I will add tests to cover this.

**Pull request merged**